pkg/storage/postgresql/stats.go

```diff
@@ -338,3 +338,75 @@ func (db DB) StatsMTTR(filter storage.Filter) (model.StatsMTTRSeverity, error) {
 
 	return mttrStats, nil
 }
+
+// StatsExposure returns the exposure times by diferent averages.
+func (db DB) StatsExposure(filter storage.Filter) (model.StatsExposure, error) {
+	exposureQuery, err := filterTemplate(`
+	WITH exposures as (
+        SELECT fe.finding_id, SUM(COALESCE (fe.ttr,
+        {{ if .AtDate }}
+		DATE_PART ('day', COALESCE (fe.expired_at, :atDate)-fe.found_at) * 24 +
+		DATE_PART ('hour', COALESCE (fe.expired_at, :atDate)-fe.found_at)
+        {{ else }}
+		DATE_PART ('day', COALESCE (fe.expired_at, NOW())-fe.found_at) * 24 +
+		DATE_PART ('hour', COALESCE (fe.expired_at, NOW())-fe.found_at)
+        {{ end }}
+        )) AS exposure
+        FROM finding_exposures fe
+        INNER JOIN findings f ON fe.finding_id = f.id
+        {{ if .AtDate }}
+        INNER JOIN finding_exposures fexp ON fexp.finding_id = f.id
+        {{ end }}
```

Comment:
I don't understand this conditional part from the `{{ if .AtDate }}`. The initial select is on the `finding_exposures` table, but here we are making a JOIN to that same table. Is this a mistake? Or there's something I'm missing?

Reply:
It is indeed a mistake, I was referencing another query and completely forgot we were already querying the `finding_exposures` table. Fixed in 340acf30c4c9d73fa19099e233c0202e8422fdb6.